src/Kestrel.Core/Internal/Http2/Http2Connection.cs

```diff
@@ -672,7 +672,13 @@ private void StartStream<TContext>(IHttpApplication<TContext> application)
             }
 
             _streams[_incomingFrame.StreamId] = _currentHeadersStream;
-            _ = _currentHeadersStream.ProcessRequestsAsync(application);
+            // Must not allow app code to block the connection handling loop.
+            ThreadPool.UnsafeQueueUserWorkItem(state =>
+            {
+                var (app, currentStream) = (Tuple<IHttpApplication<TContext>, Http2Stream>)state;
+                _ = currentStream.ProcessRequestsAsync(app);
+            },
+            new Tuple<IHttpApplication<TContext>,Http2Stream>(application, _currentHeadersStream));
```

Comment:
Nit: missing space after `>,`.

Also if we really want to avoid an allocation here, we could just add a reference to the application in Http2Stream.

Reply:
That generic is very viral. Right now it's only passed through method signatures so it can't be stored in any member fields.

Reply:
You could always cast from object in the callback, but I wouldn't bother.

Reply:
> Nit: missing space after `>,`.

Reminder :wink: